public/app/features/dashboard-scene/sharing/SharePanelEmbedTab.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import { SceneComponentProps, sceneGraph, SceneObjectBase, SceneObjectRef, VizPanel } from '@grafana/scenes';
+import { t } from 'app/core/internationalization';
+import { ShareEmbed } from 'app/features/dashboard/components/ShareModal/ShareEmbed';
+
+import { DashboardScene } from '../scene/DashboardScene';
+import { getPanelIdForVizPanel } from '../utils/utils';
+
+import { SceneShareTabState } from './types';
+
+export interface SharePanelEmbedTabState extends SceneShareTabState {
+  panelRef: SceneObjectRef<VizPanel>;
+  dashboardRef: SceneObjectRef<DashboardScene>;
+}
+
+export class SharePanelEmbedTab extends SceneObjectBase<SharePanelEmbedTabState> {
+  static Component = SharePanelEmbedTabRenderer;
+
+  public constructor(state: SharePanelEmbedTabState) {
+    super(state);
+  }
+
+  public getTabLabel() {
+    return t('share-modal.tab-title.panel-embed', 'Embed');
+  }
+}
+
+function SharePanelEmbedTabRenderer({ model }: SceneComponentProps<SharePanelEmbedTab>) {
+  const { panelRef, dashboardRef } = model.useState();
+  const p = panelRef.resolve();
+
+  const dash = dashboardRef.resolve();
+  const { uid: dashUid } = dash.useState();
```

Comment:
you don't need to use useState here, the uid will not change while this modal is open